File: nativeplugin/ios/TrueDepthFrameProcessor.m
#if __has_include(<VisionCamera/FrameProcessorPlugin.h>)
#import <VisionCamera/FrameProcessorPlugin.h>
#import <VisionCamera/FrameProcessorPluginRegistry.h>

#import "Frontend-Swift.h"

// // Example for a Swift Frame Processor plugin automatic registration
VISION_EXPORT_SWIFT_FRAME_PROCESSOR(TrueDepthFrameProcessor, trueDepthFrameProcessor)

#endif

File: nativeplugin/ios/TrueDepthFrameProcessor.swift
import  VisionCamera
import Foundation

@objc(TrueDepthFrameProcessor)
public class TrueDepthFrameProcessor: FrameProcessorPlugin {
  @objc
  public static func callback(_ frame: Frame!, withArguments arguments: [Any]!) -> Any? {
    // Your native Swift logic goes here
    print("Received frame: \(frame.width) x \(frame.height)")
    
    // Return data back to JS if needed
    return ["width": frame.width, "height": frame.height]
  }
}